jupyterlab/handlers/ydoc_handler.py

```diff
@@ -29,9 +29,9 @@ def __init__(self, type):
 
 class JupyterWebsocketServer(WebsocketServer):
     def get_room(self, path: str) -> JupyterRoom:
-        file_format, file_type, file_path = path.split(":", 2)
+        file_format, file_type, file_path, ydoc_type = path.split(":", 3)
         if path not in self.rooms.keys():
-            self.rooms[path] = JupyterRoom(file_type)
+            self.rooms[path] = JupyterRoom(ydoc_type)
```

Comment:
I don't quite understand what other information `ydoc_type` provides, compared to `file_type`?
I thought that what we wanted was for the front-end to send a configurable `file_type`, that is currently hard-coded to either `file` or `notebook`.
Am I missing something?

Reply:
Currently, `jupyter-server` only serves *directory*, *file*, and  *notebook* `file_type` (https://github.com/jupyter-server/jupyter_server/blob/7b5815e0176c079d17c539c3baf6b497197f58a7/jupyter_server/services/contents/handlers.py#L101), if we accept any `file_type`, we will need to update the content handler and managers of `jupyter-server`. I don't know enough about `jupyter-server` to judge the relevance of these changes so I went this way for the `ydoc` type.

Reply:
But you might be introducing a new parameter which is redundant with an already existing one.
Do you see `ydoc_type` taking a different value than `file_type`, if we could configure `file_type` to not be restricted to *directory*, *file* or *notebook*?
Maybe a document could be a *notebook* (for its `file_type`) but it's Y type not necessarily a *YNotebook* (for its `ydoc_type`), for instance. That would be a valid situation.

Reply:
> Yes, we can open the same file extension in different views by adding widget factories for this file extension, and in each widget factory, we can define the document model with DocumentRegistry.IModelFactory interface. By adding yDocumentType to this interface we can have different Y-types for the same file extension.

Yes, but this is what the Model does. At the moment we define one model (schema) to load the data, and then we can use this model in multiple widgets. I believe the WidgetFactory doesn't specify a `file_type` but the model's name.
I'm with David here, having a new attribute `ydoc_type` is redundant. I think we should make the actual attribute `file_type` more flexible and use it to specify the schema of the document.

Reply:
Thanks @hbcarlos for the clarification, I agree that a fix on `jupyter_server` would be a better approach. I had a direct discussion with David that leads to https://github.com/jupyter-server/jupyter_server/issues/884.